pkg/cliplugins/workspace/cmd/workspace.go

```diff
@@ -0,0 +1,570 @@
+	currentContext := o.startingConfig.Contexts[o.startingConfig.CurrentContext]
+
+	workspaceDirectoryContext := o.startingConfig.Contexts[o.workspaceDirectoryOverrides.CurrentContext]
+	workspaceDirectoryCluster := o.startingConfig.Clusters[o.workspaceDirectoryOverrides.CurrentContext]
+	if workspaceDirectoryContext == nil || workspaceDirectoryCluster == nil {
+		if currentContext != nil {
+			workspaceDirectoryCluster = o.startingConfig.Clusters[currentContext.Cluster].DeepCopy()
+			workspaceDirectoryContext = &api.Context{
+				Cluster: o.workspaceDirectoryOverrides.CurrentContext,
+			}
+		} else {
+			workspaceDirectoryCluster = &api.Cluster{}
+			workspaceDirectoryContext = &api.Context{
+				Cluster: o.workspaceDirectoryOverrides.CurrentContext,
+			}
+		}
+		workspaceDirectoryCluster.Server = o.workspaceDirectoryOverrides.ClusterInfo.Server
+		if o.workspaceDirectoryOverrides.ClusterInfo.CertificateAuthority != "" {
+			workspaceDirectoryCluster.CertificateAuthority = o.workspaceDirectoryOverrides.ClusterInfo.CertificateAuthority
+		}
+		if o.workspaceDirectoryOverrides.ClusterInfo.TLSServerName != "" {
+			workspaceDirectoryCluster.TLSServerName = o.workspaceDirectoryOverrides.ClusterInfo.TLSServerName
+		}
+		if o.workspaceDirectoryOverrides.ClusterInfo.InsecureSkipTLSVerify {
+			workspaceDirectoryCluster.InsecureSkipTLSVerify = o.workspaceDirectoryOverrides.ClusterInfo.InsecureSkipTLSVerify
+		}
+		o.startingConfig.Clusters[o.workspaceDirectoryOverrides.CurrentContext] = workspaceDirectoryCluster
+		o.startingConfig.Contexts[o.workspaceDirectoryOverrides.CurrentContext] = workspaceDirectoryContext
+
+		return clientcmd.ModifyConfig(o.configAccess, *o.startingConfig, true)
+	}
+	return nil
+}
+
+func PrioritizedAuthInfo(values ...*api.AuthInfo) *api.AuthInfo {
+	for _, value := range values {
+		if value == nil {
+			continue
+		}
+		value := *value
+		if value.Token != "" || value.TokenFile != "" || value.Password != "" || value.Username != "" {
+			return &value
+		}
+	}
+	return api.NewAuthInfo()
+}
+
+func (o *WorkspaceOptions) WorkspaceDirectoryRestConfig() (*rest.Config, error) {
+	if err := o.ensureWorkspaceDirectoryContextExists(); err != nil {
+		return nil, err
+	}
+
+	currentContextAuthInfo := &api.AuthInfo{}
+	currentContext := o.startingConfig.Contexts[o.startingConfig.CurrentContext]
+	if currentContext != nil {
+		currentContextAuthInfo = o.startingConfig.AuthInfos[currentContext.AuthInfo]
+	}
+
+	overrides := &clientcmd.ConfigOverrides{
+		AuthInfo:       *PrioritizedAuthInfo(&o.workspaceDirectoryOverrides.AuthInfo, &o.kubectlOverrides.AuthInfo, currentContextAuthInfo),
+		ClusterInfo:    o.workspaceDirectoryOverrides.ClusterInfo,
+		CurrentContext: o.workspaceDirectoryOverrides.CurrentContext,
+	}
+	return clientcmd.NewDefaultClientConfig(*o.startingConfig, overrides).ClientConfig()
+}
+
+func getScopeAndName(workspaceKey string) (string, string) {
```

Comment:
This is a pretty heavy-weight file. Do we want any part of it unit-tested?

Reply:
and security sensitive.

Reply:
also needs go doc